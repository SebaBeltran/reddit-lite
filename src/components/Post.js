import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

function Post(props) {
  console.log(props);
  const {
    author,
    title,
    permalink,
    thumbnail,
    url,
    num_comments,
    created_utc
  } = props.post;

  return (
    <div className="post_container">
      <small>
        Posted by {author} {getTimeString(created_utc)}
      </small>
      <a href={url} target="_blank" rel="noopener noreferrer">
        <h2>{title}</h2>
      </a>
      {thumbnail !== "self" && thumbnail !== "default" ? (
        <img src={thumbnail} alt="" />
      ) : null}
      <a
        href={`http://www.reddit.com${permalink}`}
        target="_blank"
        rel="noopener noreferrer"
      >
        {num_comments} comments
      </a>
    </div>
  );
}

Post.propTypes = {
  post: PropTypes.object.isRequired
};
export default Post;

export function getTimeString(time) {
  return moment.unix(time).fromNow();
}
