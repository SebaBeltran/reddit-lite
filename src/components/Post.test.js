import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Post from "./Post";

const timeMock = Math.round(new Date().getTime() / 1000);
const postMock = {
  author: "Author Name",
  title: "Post Title",
  permalink: "Permalink",
  thumbnail: "https://bit.ly/2tmUlty",
  url: "url",
  num_comments: 345,
  created_utc: timeMock
};
describe("Post", () => {
  it("displays data correctly", () => {
    const wrapper = shallow(<Post post={postMock} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
