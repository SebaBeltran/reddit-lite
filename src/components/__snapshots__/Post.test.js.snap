// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post displays data correctly 1`] = `
<div
  className="post_container"
>
  <small>
    Posted by 
    Author Name
     
    a few seconds ago
  </small>
  <a
    href="url"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h2>
      Post Title
    </h2>
  </a>
  <img
    alt=""
    src="https://bit.ly/2tmUlty"
  />
  <a
    href="http://www.reddit.comPermalink"
    rel="noopener noreferrer"
    target="_blank"
  >
    345
     comments
  </a>
</div>
`;
